src/main/java/org/logstash/beats/CompressedBatchEncoder.java

```diff
@@ -0,0 +1,40 @@
+package org.logstash.beats;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.channel.ChannelHandlerContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import java.io.IOException;
+import java.util.zip.Deflater;
+import java.util.zip.DeflaterOutputStream;
+
+public class CompressedBatchEncoder extends BatchEncoder {
+    private final static Logger logger = LogManager.getLogger(BatchEncoder.class.getName());
+
+    @Override
+    protected ByteBuf getPayload(ChannelHandlerContext ctx, Batch batch) throws IOException {
+        ByteBuf payload = super.getPayload(ctx, batch);
+
+        Deflater deflater = new Deflater();
+        ByteBufOutputStream output = new ByteBufOutputStream(ctx.alloc().buffer());
```

Comment:
again try-with-resources here to ensure that every resource gets closed, no matter if exceptions are thrown